File: backend/src/main/java/com/fullteaching/backend/model/ChatConversation.java
package com.fullteaching.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.Collection;
import java.util.HashSet;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode
public class ChatConversation {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @ManyToMany
    private Collection<User> users = new HashSet<>();

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, orphanRemoval = true)
    private Collection<ChatMessage> messages = new HashSet<>();
}

File: frontend/src/app/classes/chat-conversation.ts
import {User} from './user';
import {ChatMessage} from './chat-message';

export class ChatConversation {

  id: number;
  users: User[];
  messages: ChatMessage[];

  constructor(id: number, users: User[]) {
    this.id = id;
    this.users = users;
    this.messages = [];
  }
}
